readinglist/tests/test_views_article.py

```diff
@@ -85,24 +86,129 @@ def test_string_filters_naively_by_value(self):
+        modification = resp.json['last_modified']
+        header = float(resp.headers['Timestamp'])
+        self.assertEqual(header, modification)
+
+    def test_filter_with_since_accepts_numeric_value(self):
+        article = MINIMALIST_ARTICLE.copy()
+        self.app.post_json('/articles', article, headers=self.headers)
+        url = '/articles?_since=6'
+        resp = self.app.get(url, headers=self.headers)
+        self.assertEqual(len(resp.json['items']), 1)
+
+    def test_filter_from_last_modified_is_exclusive(self):
+        article = MINIMALIST_ARTICLE.copy()
+        resp = self.app.post_json('/articles', article, headers=self.headers)
+
+        current = resp.json['last_modified']
+        url = '/articles?_since={0}'.format(current)
+
+        resp = self.app.get(url, headers=self.headers)
+        self.assertEqual(len(resp.json['items']), 0)
+
+    def test_filter_from_last_header_value_is_exclusive(self):
+        article = MINIMALIST_ARTICLE.copy()
+        resp = self.app.post_json('/articles', article, headers=self.headers)
+
+        current = float(resp.headers['Timestamp'])
+        url = '/articles?_since={0:.3f}'.format(current)
+
+        resp = self.app.get(url, headers=self.headers)
+        self.assertEqual(len(resp.json['items']), 0)
+
     def test_filter_works_with_empty_list(self):
         self.fxa_verify.return_value = {
             'user': 'jean-louis'
         }
         self.app.get('/articles?unread=true', headers=self.headers)
 
+    def test_timestamp_are_always_identical_on_read(self):
+
+        def read_timestamp():
+            resp = self.app.head('/articles', headers=self.headers)
+            return int(resp.headers['Timestamp'])
+
+        before = read_timestamp()
+        now = read_timestamp()
+        after = read_timestamp()
+        self.assertTrue(before == now == after)
+
+    def test_timestamp_are_always_incremented_on_creation(self):
+
+        def read_timestamp():
+            resp = self.app.post_json('/articles',
+                                      MINIMALIST_ARTICLE,
+                                      headers=self.headers)
+            return int(resp.json['last_modified'])
+
+        before = read_timestamp()
+        now = read_timestamp()
+        after = read_timestamp()
+        self.assertTrue(before < now < after)
+
+    def test_records_created_during_fetch_are_above_fetch_revision(self):
+
+        revisions = {}
+
+        def long_fetch():
+
+            def delayed_get(*args, **kwargs):
+                time.sleep(.25)
+                return MINIMALIST_ARTICLE
+
+            with mock.patch.object(self.db, 'get_all', delayed_get):
+                resp = self.app.head('/articles',
+                                     headers=self.headers)
+                revisions['fetch'] = resp.headers['Timestamp']
+
+        thread = threading.Thread(target=long_fetch)
+        thread.start()
+        resp = self.app.post_json('/articles',
```

Comment:
if this line fails, you are leaving a thread running. use try..finally to perform the .join()